trio/testing/_fake_net.py

```diff
@@ -176,7 +193,7 @@ def __init__(self, fake_net: FakeNet, family: int, type: int, proto: int):
 
         self._closed = False
 
-        self._packet_sender, self._packet_receiver = trio.open_memory_channel(
+        self._packet_sender, self._packet_receiver = trio.open_memory_channel[object](
```

Comment:
Using object here (and the other similar calls) isn't really right, it'll cause problems later when we type these functions further?